Core/include/Acts/Vertexing/VertexingOptions.hpp

```diff
@@ -24,22 +24,44 @@ struct VertexingOptions {
   /// Default constructor is deleted
   VertexingOptions() = delete;
 
-  /// VertexingOptions with context
+  /// VertexingOptions with context and beam spot constraint
   ///
   /// @param gctx The goemetry context for this fit
   /// @param mctx The magnetic context for this fit
-  /// @param vconstr The pointing constraint to a vertex
-  VertexingOptions(
-      const GeometryContext& gctx, const MagneticFieldContext& mctx,
-      const Vertex<input_track_t>& vconstr = Vertex<input_track_t>())
-      : geoContext(gctx), magFieldContext(mctx), vertexConstraint(vconstr) {}
+  /// @param bsConstr The beamspot constraint
+  /// @param useConstr Boolean indicating whether vertex constraint should be used
+  VertexingOptions(const GeometryContext& gctx,
+                   const MagneticFieldContext& mctx,
+                   const Vertex<input_track_t>& bsConstr,
+                   const bool useConstr = true)
```

Comment:
Could this be something like

```suggestion
                   std::optional<Vertex<input_track_t>> bsConstr)
```

Maybe we could get rid of the other constructors in that case.

Reply:
Unfortunately, the beam spot constraint is always used as the initial guess of the vertex position. Therefore, it needs to be set!
I guess the name of the boolean is kind of misleading, should I change it? If so, do you have an idea?